### ISSUE TITLE
Are the classification step and self-expression matrices trained separately?

### ISSUE THREAD

**Issue by dugzzuli** [buschju/sscn#2]
Hello!

After reading the code of this warehouse, I have a little doubt. Is this classification step and self-expression matrix trained in two steps?

# Forward pass
Z = self.encoder(X)
H = self.se_encoder(Z).detach() We note that this place executes the detach operation and disconnects the relationship with the above network. And there is no optimizer expression loss and reconstruction loss.

for _ in range(num_iterations_per_batch):
    outputs = self.classifier.get_outputs(H)
    loss_assignment = get_loss_supervised(outputs=outputs,
                                          y=pseudo_labels,
                                          )

    # Backward pass
    optimizer.zero_grad()
    loss_assignment.backward()
    optimizer.step()
